silver/models/payment_methods.py

```diff
@@ -84,6 +89,35 @@ def decrypt_data(self, crypted_data):
         except InvalidToken:
             return None
 
+    def cancel(self):
+        if self.canceled:
+            raise ValidationError("You can't cancel a canceled payment method.")
+
+        cancelable_states = [Transaction.States.Initial,
+                             Transaction.States.Pending]
+        transactions = Transaction.objects.filter(payment_method=self,
```

Comment:
You don't have to import Transaction for this. There probably is a related_name for the reverse relation on which can further filter.